pkg/fileutil/fileutil.go

```diff
@@ -79,6 +80,10 @@ func EnsureExecutable(p string) error {
 	if err != nil {
 		return fmt.Errorf("error doing stat on %q: %v", p, err)
 	}
+	m := s.Mode()
+	if m&(syscall.S_IXOTH|syscall.S_IXGRP|syscall.S_IXUSR) != 0 {
+		return nil
+	}
 	if err := os.Chmod(p, s.Mode()|0111); err != nil {
```

Comment:
Nice use of `syscall.S_IXOTH|syscall.S_IXGRP|syscall.S_IXUSR`! Should we do 
```
os.Chmod(p, s.Mode()|syscall.S_IXOTH|syscall.S_IXGRP|syscall.S_IXUSR)
``` 
as well instead of using `0111`?

Reply:
Curious what's the harm if chmod an executable again? From the error message below from the linked issue, it seems a permission issue to me. Could it because the `aws` is owned by `root` while `aws-k8s-tester` is running as some other user?

```
failed to create EKS deployer error doing chmod on "/usr/local/bin/aws": chmod /usr/local/bin/aws: operation not permitted
```

Reply:
Yes, that's precisely what is the case. /usr/local/bin/aws is owned by root for some reason. But the point of this patch is that EnsureExecutable is merely looking to ensure the aws CLI tool is executable, nothing more, and chmod'ing a file that is owned by root isn't necessary to check for execute permissions.